### PR TITLE
Introspect views

### LINKED ISSUE #270: Support introspection of materialized views
As I was investigating how to implement the [type generation for database views](https://github.com/RobinBlomberg/kysely-codegen/issues/53) I found out that the [MysqlIntrospector](https://github.com/koskimas/kysely/blob/master/src/dialect/mysql/mysql-introspector.ts) already returns views on a `getTables` call.

If all introspectors would behave in this way the feature would be automatically present in kysely-codegen.

Would there be any interest in merging a PR that makes postgres and sqlite `getTables` return database views?

### PR DESCRIPTION
**Pull request by diogob**
Closes #270 

Ensure all instrospector return database views and add a boolean isView to the TableMetadata so users can tell them apart.

## Todo
- [x] document

### CODE REVIEW

**Comment by diogob** [comment]
I'm not sure on how to proceed on the documentation front:

* Regarding the generated documentation, if I regenerate it running `npm run docs:build` it replaces several references to commits and creates a big diff. Therefore I was unsure if I should include it in the PR, please advise.
* Regarding recipes or examples, I could not find any reference to the `getTables` method, would it be interesting to write something in this PR?

**Comment by igalklebanov** [comment]
Thank you! 💪 

> * Regarding the generated documentation, if I regenerate it running `npm run docs:build` it replaces several references to commits and creates a big diff. Therefore I was unsure if I should include it in the PR, please advise.

You can leave that out. @koskimas usually preps everything for release, we might automate this in the future in pre-publish with some other steps.

> * Regarding recipes or examples, I could not find any reference to the `getTables` method, would it be interesting to write something in this PR?

Generally, most documentation is written inline in ts docs. If you feel like subject at hand demands a recipe or a section in the main README.md, feel free to add that. Maybe an introspection recipe?

File: src/dialect/mysql/mysql-introspector.ts [hunk 1]
```diff
@@ -35,24 +35,30 @@ export class MysqlIntrospector implements DatabaseIntrospector {
...
+      .where('columns.table_schema', '=', sql`database()`)
+      .orderBy('columns.table_name')
+      .orderBy('columns.ordinal_position')
       .$castTo<RawColumnMetadata>()
 
     if (!options.withInternalKyselyTables) {
       query = query
-        .where('table_name', '!=', DEFAULT_MIGRATION_TABLE)
-        .where('table_name', '!=', DEFAULT_MIGRATION_LOCK_TABLE)
+        .where('columns.table_name' as string, '!=', DEFAULT_MIGRATION_TABLE)
```
Comment (koskimas) [inline src/dialect/mysql/mysql-introspector.ts:60]: Is the explicit cast to string really needed here?

**Review by koskimas — APPROVED** [review]

**Comment by koskimas** [comment]
Awesome work @diogob! Thank you ❤️

**Pull request merged**